bin/improver-nbhood-iterate-with-mask

```diff
@@ -112,7 +116,22 @@ def main():
                              're-masking is applied to the neighbourhood '
                              'processed output. Therefore, the neighbourhood '
                              'processing may result in values being present '
-                             'in areas that were originally masked. ')
+                             'in areas that were originally masked. This '
+                             'allows the the values in adjacent bands to be'
+                             'weighted together if the additional dimension'
+                             'from the masking process is collapsed.')
+    parser.add_argument('--collapse_dimension', action='store_true',
+                        help='Collapse the dimension from the mask, by doing '
+                             'a weighted mean using the weights provided.')
+    parser.add_argument('--weights_for_collapsing_dim', metavar='WEIGHTS',
+                        default=None,
+                        help='The weights used for collapsing the dimension '
+                             'gained through masking. If they are not '
+                             'provided they default to equal weights.')
```

Comment:
You need to specify that WEIGHTS is the path to a NetCDF file.

Reply:
Added